### DISCUSSION TITLE
curl_global_init must be called?

### DISCUSSION THREAD

**Discussion by maoxing** [curl/curl#12712] [Q&A] [answered]
according to the doc

curl_global_init function must be called at least once within a program (a program is all the code that shares a memory space) before the program calls any other function in libcurl. The environment it sets up is constant for the life of the program and is the same for every program, so multiple calls have the same effect as one call.

however, I did not call it manually in the program and have not found any problems so far. Are there any hidden dangers?

**Comment by icing** [comment, marked as answer]
Yes.

**Reply by maoxing** [reply]
i'm not call it, but my program is running fine!

**Reply by icing** [reply]
Maybe so. But in the next curl version, we might add something in the init that is necessary on your platform.

**Reply by maoxing** [reply]
so I'm confused,  version is 7.52.1

Got it, Thanks reply

**Comment by jay** [comment]
It's working for you because [curl_easy_init](https://curl.se/libcurl/c/curl_easy_init.html) calls curl_global_init(CURL_GLOBAL_DEFAULT) as a fallback when you haven't already called curl_global_init on your own. Technically you should call [curl_global_init](https://curl.se/libcurl/c/curl_global_init.html) before any other libcurl function, with the exception of [curl_global_sslset](https://curl.se/libcurl/c/curl_global_sslset.html) to select an SSL library before initialization.
